src/github/client/github-client.ts

```diff
@@ -98,17 +98,19 @@ export default class GitHubClient {
 		return response;
 	}
 
-	private async graphql<T>(query: string): Promise<AxiosResponse> {
-		return await this.axios.post<T>("https://api.github.com/graphql",
+	public async graphql<T>(query: string, variables: any): Promise<AxiosResponse<T>> {
```

Comment:
Variables should be optional or have a default.  Also, don't use 'any', should probably be something like `Record<string, string | number>`

Reply:
This should stay private as we don't want people to use the graphql querying directly, we want them to use a 'getBranch' function which calls the graphql function.

Reply:
I think keeping it `private` defeats the purpose of having a graphql option. Keeping it private, everytime we would want add new code to client even if there is a small difference in query.

Reply:
I don't agree that we shouldn't expose graphql option. GraphQL queries can be complex and tied directly to a business logic. Some GraphQL queries are not generic hence should be the responsibility of the business logic to control them, not the client.

Reply:
It's not a client, it's a service.  Axios is the client.  This already has business logic to it.

The whole point of having this service is to centralize common functionality, like retrieving branches, and make sure their type safe.  If you expose the graphql function, you lose both.

Reply:
Aren't the name says 'github-client'?

So, for normal APIs what you suggested was working and we were doing it in this way. For GraphQL it is not really the case for the reason explained above. 

Right now the change is simple, elegant and easily feature flaggable. Should we make it complicated and increase the risk of breaking Initial Sync?

Reply:
I agree that the `graphql()` function should stay private and we instead should expose use-case-specific functions like `getBranch()` etc that use it under the hood. Code should be as specific as possible and only as generic as it has to be.

Reply:
Here's what the trade off will be, the current query is specific for things you need for initial sync. In case, you want to use `getBranches()` somewhere else, and it doesnt have the specific items in the query, the query will need to be updated or we would have to add another function like `getBranchDetailsForSomething()`

I think there is no right/wrong way here its more like what are we ready to change the query and test everything when something changes ?

Reply:
Yeah, I'd go for multiple `getBranch` functions as needed if they're that different.  The whole point of the service is to abstract out what's happening behind the covers; is it using REST or GraphQL? The code that calls it doesn't know and shouldn't care.  What it should care about is what are the inputs needed for this function and the type that it spits out afterwards.  

It makes the whole thing much more testable, consistent and stable.

Reply:
No worries, will make the change 👍